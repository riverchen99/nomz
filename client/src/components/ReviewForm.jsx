import React from 'react';
import { StyledReviewForm, TextInput, SubmitComponent } from './StyledReviewForm';
import Button from './Button';

class ReviewForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.props.onReviewChange(event.target.value);
  }

  handleSubmit(event) {
    this.props.onReviewSubmit();
    event.preventDefault();
  }

  render() {
    return (
      <StyledReviewForm onSubmit={this.handleSubmit}>
        <TextInput 
          type="text" 
          value={this.props.review_text} 
          onChange={this.handleChange} 
          rows={8} 
          required
          placeholder={"Write your review here!"}/>
        <SubmitComponent>
          Your name and profile picture will be shown on the review.
          <Button type="submit" text="Post Review" color="rgb(239, 57, 255)" />
        </SubmitComponent>
      </StyledReviewForm>
    );
  }
}

export default ReviewForm;
