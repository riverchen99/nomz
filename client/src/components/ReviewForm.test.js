import React from 'react';
import { shallow, mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import ReviewForm from './ReviewForm';

configure({ adapter: new Adapter() });

const reviewChangeFn = jest.fn();
const submitFn = jest.fn();
describe('Input to submit text review', () => {
  it('should render correctly without props', () => {
    const component = shallow(<ReviewForm />);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly with onReviewSubmit and onReviewChange', () => {
    const component = mount(<ReviewForm onReviewSubmit={submitFn} onReviewChange={reviewChangeFn} />);
    expect(component).toMatchSnapshot();
  });
  it('calls the onReviewSubmit function correctly', () => {
    const component = mount(<ReviewForm onReviewSubmit={submitFn} onReviewChange={reviewChangeFn} />);
    component.props().onReviewSubmit();
    expect(submitFn).toHaveBeenCalled();
  });
  it('calls the onReviewChange function correctly', () => {
    const component = mount(<ReviewForm onReviewSubmit={submitFn} onReviewChange={reviewChangeFn} />);
    component.props().onReviewChange();
    expect(reviewChangeFn).toHaveBeenCalled();
  });
});
