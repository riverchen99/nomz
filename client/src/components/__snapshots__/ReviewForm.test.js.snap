// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input to submit text review should render correctly with onReviewSubmit and onReviewChange 1`] = `ReactWrapper {}`;

exports[`Input to submit text review should render correctly without props 1`] = `ShallowWrapper {}`;
